app/src/main/java/com/pysiak/simple/clock/ui/clock/ClockPresenter.java

```diff
@@ -8,8 +14,19 @@
 public class ClockPresenter {
 
     private ClockView clockView;
+    private static final String BASE_URL = "http://api.timezonedb.com";
+    private Retrofit retrofit;
+    private TimezonedbApi timezonedbApi;
+
+    private Context context;
 
-    ClockPresenter() {
+    ClockPresenter(Context context) {
+        this.context = context;
+        retrofit = new Retrofit.Builder()
```

Comment:
Move retrofit and all related calls/constants to Application class. This is what you have seen in all examples, I've no idea why you've skipped such realization